src/servicehandler.cpp

```diff
@@ -194,122 +92,171 @@ ServiceHandler::ServiceHandler()
+                interface = service(defaultName);
+            }
+            d->mError = interface ? (ServiceHandler::ErrorCode)interface->error() : ErrorInternal;
+            return false;
+        }
+    } else {
         return false;
-
-    return d->executePlugin(SendResponse, invitation, body, calendar, storage, notebook);
+    }
 }
 
-QString ServiceHandler::icon(const Notebook::Ptr &notebook, const ExtendedStorage::Ptr &storage)
+QString ServiceHandler::icon(const QString &serviceId)
 {
-    if (storage.isNull() || notebook.isNull())
-        return QString();
-
-    ServiceInterface *service = d->getServicePlugin(notebook, storage);
+    ServiceInterface *plugin = service(serviceId);
+    if (!plugin) {
+        plugin = service(defaultName);
+    }
 
-    if (service) {
-        QString res(service->icon());
+    d->mError = ErrorOk;
+    if (plugin) {
+        QString res(plugin->icon());
         if (res.isNull()) {
-            d->mError = (ServiceHandler::ErrorCode) service->error(); //Right now convert directly
+            d->mError = (ServiceHandler::ErrorCode) plugin->error(); //Right now convert directly
         }
         return res;
     } else {
         return QString();
     }
 }
 
-bool ServiceHandler::multiCalendar(const Notebook::Ptr &notebook, const ExtendedStorage::Ptr &storage)
+bool ServiceHandler::multiCalendar(const QString &serviceId)
 {
-    if (storage.isNull() || notebook.isNull())
-        return false;
-
-    ServiceInterface *service = d->getServicePlugin(notebook, storage);
+    ServiceInterface *plugin = service(serviceId);
+    if (!plugin) {
+        plugin = service(defaultName);
+    }
 
-    if (service) {
-        bool res = service->multiCalendar();
-        if (!res) {
-            d->mError = (ServiceHandler::ErrorCode) service->error(); //Right now convert directly
-        }
+    d->mError = ErrorOk;
+    if (plugin) {
+        bool res = plugin->multiCalendar();
+        d->mError = (ServiceHandler::ErrorCode) plugin->error(); //Right now convert directly
         return res;
     } else {
         return false;
     }
 }
 
-QString ServiceHandler::emailAddress(const Notebook::Ptr &notebook, const ExtendedStorage::Ptr &storage)
+QString ServiceHandler::emailAddress(const Notebook::Ptr &notebook)
 {
-    if (storage.isNull() || notebook.isNull())
-        return QString();
-
-    ServiceInterface *service = d->getServicePlugin(notebook, storage);
+    ServiceInterface *plugin = service(notebook->pluginName());
+    if (!plugin) {
+        plugin = service(defaultName);
+    }
 
-    if (service) {
-        QString res = service->emailAddress(notebook);
+    d->mError = ErrorOk;
+    if (plugin) {
+        QString res = plugin->emailAddress(notebook);
         if (res.isNull()) {
-            d->mError = (ServiceHandler::ErrorCode) service->error(); //Right now convert directly
+            d->mError = (ServiceHandler::ErrorCode) plugin->error(); //Right now convert directly
```

Comment:
Was already like this, but the null QString is not defined to be any indication of error, and just returning a QString() is quite conventional way to return an empty QString. Moreover as mentioned we don't need error handlings for plain property getters so let's remove this and the mError assignments a few lines above.

Applies to couple other similar cases around too.

Reply:
Ok, I see. I've removed the error handling for `icon`, `emailAddress` and `displayName` getters.